src/controllers/posts_controller.ts

```diff
@@ -0,0 +1,72 @@
+import { Request, Response } from "express";
+import { handleMongoQueryError } from "../../db";
+import Post, { PostDocument } from "../models/posts_model";
+
+export const getPosts = async (req: Request, res: Response): Promise<any> => {
+  const { sender }: { sender?: string } = req.query;
+
+  try {
+    let posts: PostDocument[] | null = await (sender ? Post.find({ sender: sender }) : Post.find());
```

Comment:
I don't think it can return null. When nothing is found it should be an empty array.